indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -94,83 +104,68 @@
-                class: 'js-search-category',
                 type: 'search',
-                name: 'q',
                 placeholder: $T.gettext("Search")
-            }).attr('autocomplete', 'off');
+            });
             self.element.prepend(self.$searchInput);
-
-            self.$searchInput.on('input', _.debounce(function(evt) {
-                var $this = $(this);
-                var inputValue = $this.val().trim();
-
-                if (!inputValue) {
-                    self._clearSearch();
-                }
-
-                if (inputValue.length < 3) {
-                    return;
-                }
-
-                self._currentSearchRequest = $.ajax({
-                    url: build_url(Indico.Urls.Categories.search),
-                    data: {q: inputValue},
-                    beforeSend: function() {
-                        if (self._currentSearchRequest != null) {
-                            self._currentSearchRequest.abort();
-                        }
-                        self._toggleLoading(true);
-                    },
-                    complete: function() {
-                        self._toggleLoading(false);
-                    },
-                    error: function(jqXHR) {
-                        if (jqXHR.statusText === 'abort') {
-                            return;
-                        }
-                        handleAjaxError(jqXHR);
-                    },
-                    success: function(data) {
-                        if (self._isInDOM()) {
-                            self.$category.hide();
-                            self.$categoryTree.hide();
-                            self._renderSearchResultInfo(data.categories.length, data.total_count);
-                            self._renderSearchResultList(data.categories);
-                        }
+            self.$searchInput.realtimefilter({
+                wait: 500,
+                callback: function(value) {
+                    if (!value) {
+                        self._clearSearch();
+                    } else {
+                        self.searchCategories(value);
                     }
-                });
-            }, 500));
+                }
+            });
         },
 
-        _clearSearch: function() {
+        _createBindings: function() {
             var self = this;
-
-            if (self._currentSearchRequest != null) {
-                self._currentSearchRequest.abort();
-            }
-            self.$searchInput.val('');
-            self.$categoryResults.hide();
-            self.$categoryResultsList.empty();
-            self.$categoryResultsInfo.empty();
-            self.$category.show();
-            self.$categoryTree.show();
+            self.element.on('click', '.js-action', function(evt) {
+                var $this = $(this);
+                evt.stopPropagation();
+                if ($this.hasClass('disabled')) {
+                    evt.preventDefault();
```

Comment:
Truth is we don't need it at all since we only apply `disabled` on a `<span>` element.